Remotion/Web/Development.WebTesting/CoypuElementScopeExtensions.cs

```diff
@@ -40,27 +40,56 @@ public static class CoypuElementScopeExtensions
     /// where the <c>parentScope.Find*()</c> call could not be found.
     /// </summary>
     /// <param name="scope">The <see cref="ElementScope"/> which is asserted to exist.</param>
+    /// <exception cref="WebTestException">If the control cannot be found.</exception>
+    /// <exception cref="WebTestException">If multiple matching controls are found.</exception>
     public static void EnsureExistence ([NotNull] this ElementScope scope)
     {
       ArgumentUtility.CheckNotNull ("scope", scope);
 
-      scope.Now();
+      try
+      {
+        scope.Now();
+      }
+      catch (MissingHtmlException exception)
+      {
+        throw AssertionExceptionUtility.CreateControlMissingException (exception);
+      }
+      catch (AmbiguousException exception)
+      {
+        throw AssertionExceptionUtility.CreateControlAmbiguousException (exception);
+      }
     }
 
     /// <summary>
     /// Ensures that the given <see cref="Scope"/> exists, much like <see cref="EnsureExistence"/>. However, it ensures that Coypu uses the
     /// <see cref="Match.Single"/> matching strategy.
     /// </summary>
     /// <param name="scope">The <see cref="ElementScope"/> which is asserted to match only a single DOM element.</param>
+    /// <exception cref="WebTestException">If the control cannot be found.</exception>
+    /// <exception cref="WebTestException">If multiple matching controls are found.</exception>
     public static void EnsureSingle ([NotNull] this ElementScope scope)
     {
       ArgumentUtility.CheckNotNull ("scope", scope);
 
       var matchBackup = scope.ElementFinder.Options.Match;
 
-      scope.ElementFinder.Options.Match = Match.Single;
-      scope.Now();
-      scope.ElementFinder.Options.Match = matchBackup;
+      try
+      {
+        scope.ElementFinder.Options.Match = Match.Single;
+        scope.Now();
+      }
+      catch (MissingHtmlException exception)
+      {
+        throw AssertionExceptionUtility.CreateControlMissingException (exception);

```

Comment:
The method names of `CreateControlMissingException` and `CreateControlAmbiguousException` have not been changed yet. Again, I believe this is so deep within our architecture that we don't even want to throw a `WebTestException`.

Reply:
This method is the common denominator of selectors. And if a selector fails, its a WebTestException because the element cannot be found on the page.